### PR TITLE
Remove API deprecated since many releases

### PR DESCRIPTION
**Pull request by forman**
Removed API deprecated since many releases:
  - Removed keyword argument `base` from function 
    `xcube.core.resampling.temporal.resample_in_time()`.
  - Removed option `base` from CLI command `xcube resample`.
  - Removed keyword argument `assert_cube` from 
    `xcube.core.timeseries.get_time_series()`.
  - Removed property `xcube.core.xarray.DatasetAccessor.levels`.
  - Removed function `xcube.core.tile.parse_non_spatial_labels()`.
  - Removed keyword argument `tag` from context manager 
    `xcube.util.perf.measure_time()`.
  - Removed function `xcube.core.geom.convert_geometry()`.
  - Removed function `xcube.core.geom.is_dataset_y_axis_inverted()`.
  - Removed function `xcube.util.assertions.assert_condition()`.
  - Removed function `xcube.util.cmaps.get_cmaps()`.
  - Removed function `xcube.util.cmaps.get_cmap()`.
  - Removed function `xcube.util.cmaps.ensure_cmaps_loaded()`.
  - Removed endpoint `/datasets/{datasetId}/vars/{varName}/tiles2/{z}/{y}/{x}`
    from xcube server.

Checklist:

* [ ] ~Add unit tests and/or doctests in docstrings~
* [ ] ~Add docstrings and API docs for any new/modified user-facing classes and functions~
* [ ] ~New/modified features documented in `docs/source/*`~
* [x] Changes documented in `CHANGES.md`
* [ ] GitHub CI passes
* [ ] AppVeyor CI passes
* [x] Test coverage remains or increases (target 100%)

### CODE REVIEW

File: xcube/core/geom.py [hunk 1]
```diff
@@ -622,31 +621,6 @@ def normalize_geometry(
     raise ValueError(_INVALID_GEOMETRY_MSG)
 
 
-@deprecated(
-    version="0.11.2",
-    reason="convert_geometry() has been" " renamed to normalize_geometry()",
-)
-def convert_geometry(
```
Comment (konstntokas) [inline xcube/core/geom.py:629]: Some docstrings of other functions refer to convert_geometry(). See [line 300](https://github.com/xcube-dev/xcube/blob/forman-remove_deprecated_code/xcube/core/geom.py#L300) and [line 447](https://github.com/xcube-dev/xcube/blob/forman-remove_deprecated_code/xcube/core/geom.py#L447). 

Also not sure if docs need to be adjusted [here](https://github.com/xcube-dev/xcube/blob/forman-remove_deprecated_code/docs/source/api.rst?plain=1#L206).

**Review by konstntokas — CHANGES_REQUESTED** [review]
I checked all the removed functions by searching them throughout the project. Please remove `convert_geometry` in the docstrings and docs mentioned in my comment.

File: xcube/core/geom.py (hunk 1 above)
Comment (forman) [reply xcube/core/geom.py:629]: Done!

Comment (konstntokas) [reply xcube/core/geom.py:629]: Aren't the references to convert_geometry() in the docstrings in [`xcube/core/geom.py` line 300](https://github.com/xcube-dev/xcube/blob/forman-remove_deprecated_code/xcube/core/geom.py#L300) and [`xcube/core/geom.py` line 447](https://github.com/xcube-dev/xcube/blob/forman-remove_deprecated_code/xcube/core/geom.py#L447) confusing? Or am I understanding something wrong here?!

File: xcube/core/dsio.py [hunk 2]
```diff
@@ -46,6 +46,8 @@
 _DEPRECATION_VERSION = "0.12.1"
 
 
+# Note, we cannot remove deprecated code as long as
```
Comment (thomasstorm) [inline xcube/core/dsio.py:49]:
```suggestion
# Note, we cannot remove this deprecated code as long as
```

**Review by thomasstorm — APPROVED** [review]
Just a tiny hint, looks fine otherwise!

**Review by pont-us — APPROVED** [review]

**Review by konstntokas — APPROVED** [review]
Please consider my [previous coment](https://github.com/xcube-dev/xcube/pull/983#discussion_r1613193992). Otherwise approved :)

**Pull request merged**